src/applications/personalization/profile-2/components/connected-apps/AdditionalInfoSections.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect } from 'react';
+import AdditionalInfo from '@department-of-veterans-affairs/formation-react/AdditionalInfo';
+import PropTypes from 'prop-types';
+
+import { focusElement } from 'platform/utilities/ui';
+
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+import { isEmpty } from 'lodash';
+import availableConnectedApps from './availableConnectedApps';
+
+import Telephone, {
+  CONTACTS,
+} from '@department-of-veterans-affairs/formation-react/Telephone';
+
+export const AdditionalInfoSections = ({ activeApps }) => {
+  const connectedAppsNames = activeApps
+    ? activeApps.map(app => app.attributes.title)
+    : [];
+
+  // This filters out any availableConnectedApps that the user has already connected to
+  const filteredApps = availableConnectedApps.filter(
+    app => !connectedAppsNames.includes(app.name),
+  );
+
+  return (
+    <>
+      {!isEmpty(activeApps) &&
```

Comment:
Ideas for refactoring this: @rianfowler introduced me to the idea of putting this kind of logic outside of the actual render function in aptly-named vars. So you might have something like `const showWhatElseCanIAddInfo = !isEmpty(activeApps) && !isEmpty(filteredApps)` declared above this and then this part of the function reads a little nicer. The idea basically being the `return` part of the functional component should be very clean and easy to read. You could take this further by abstracting away the actual JSX for a given case. So in the end this could read like:
```
{showWhatElseCanIAdd && <WhatElseCanIAddInfo>}
```

Reply:
I 100% agree! All about that clear variable naming 👍  💯

Reply:
I did assign those conditional combinations to new variables, but left them within the render as the values need to be computed within the render